### ISSUE TITLE
EventPipe Streaming support for greater than 4GB streams

### ISSUE THREAD

**Issue by noahfalk** [dotnet/coreclr#24188]
The Netperf file format appears to have size limitations because it uses 32 bit values to refer to other positions in the stream. It doesn't look hard to generate 1-10 MB/sec of trace data which means a stream could hit the limit in under 10 minutes of operation. Alternately a long running dotnet-counters scenario might log at 10 KB/sec which will hit the stream limit in about 5 days of service uptime.

We need to:
- [x] Identify any fundamental limitations in the Netperf format so that we can theoretically support 2^64 bytes of streamed or on-disk data
- [x] Update the netperf format documentation
- [x] Update the emitter in the runtime to generate the modified format
- [ ] Update TraceEvent and PerfView to parse the modified format, as well as eliminate 32 bit limitations in that code
- [ ] Test streaming scenarios that exceed 4GB with our various tools

You could also imagine attacking this problem from the angle of compression but I think that is at best a complementary effort. Even with a 10x reduction in stream size we've only moved the 10 minute scenario to a 100 minute scenario. If we want this to be considered a serious technology I think we need to support sustained high rate telemetry for services that might run non-stop for months.

cc @jorive @sywhang @brianrob @vancem

**Comment by noahfalk** [comment]
In the upgrade to the NetTrace format I identified one issue, emitting BeginObject/EndObject tag pairs causes the FastSerialization library to add that object to an intern table that grows without bound. Technically NetTrace doesn't need that data interned so it could have been solved at the reader end without a format change, but it seemed easier to use BeginPrivateObject in the format and not add custom reading modes to FastSerialization. Other than that I didn't spot any 32 bit stream size limitations in the file format.

These changes have been implemented and documented in #24853 for coreclr and https://github.com/microsoft/perfview/pull/948 for PerfView. 

On the TraceEvent side there are still 32 bit indexes being used in the reader that need to be fixed and then testing needs to be done.
